### ISSUE TITLE
Custom .so loading breaks Android App Bundles built with Android Gradle Plugin 3.3

### ISSUE THREAD

**Issue by scottkennedy** [facebook/react-native#23149]
## Environment
```
React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.6
      CPU: x64 Intel(R) Core(TM) i7-6920HQ CPU @ 2.90GHz
      Memory: 1.23 GB / 16.00 GB
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 8.11.2 - /var/folders/m0/z43h23_j2z93cptz7ddffst4h3whf0/T/yarn--1548357696136-0.8201936393698348/node
      Yarn: 1.12.3 - /var/folders/m0/z43h23_j2z93cptz7ddffst4h3whf0/T/yarn--1548357696136-0.8201936393698348/yarn
      npm: 5.6.0 - ~/.nvm/versions/node/v8.11.2/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
      Android SDK:
        Build Tools: 23.0.1, 26.0.3, 27.0.2, 27.0.3, 28.0.2, 28.0.3
        API Levels: 23, 25, 26, 27, 28
    IDEs:
      Android Studio: 3.2 AI-181.5540.7.32.5056338
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      @types/react: 16.4.18 => 16.4.18
      @types/react-native: 0.57.7 => 0.57.7
      react: 16.6.1 => 16.6.1
      react-native: 0.57.7 => 0.57.7
    npmGlobalPackages:
      react-native-cli: 2.0.1

✨  Done in 13.49s.
```

## Description
https://issuetracker.google.com/issues/122953695#comment13 has some comments from Google on this.  The .so files are not extracted to disk, so they don't exist where RN is looking for them.  This only occurs with .aab files (app bundles), not .apk files.

"Someone should probably file a bug with reactnative so they stop reading the .so from the install directory and instead read it from the APK or use the standard Android API to load so files."

## Reproducible Demo
Build an Android App Bundle (guide at https://developer.android.com/guide/app-bundle/) using Android Gradle Plugin 3.3 and try to load React Native - it will fail to find libreactnativejni.so

https://developer.android.com/studio/run/#reference has information on App Bundles.

**Comment by dulmandakh** [comment]
App Bundling support landed in 0.58. Please upgrade to 0.58.

**Comment by danilobuerger** [comment]
@dulmandakh which commit are you referring to?

**Comment by dulmandakh** [comment]
https://github.com/facebook/react-native/commit/abfd563b471b8f57d6d6f54862082a49e1a58258

**Comment by tgensol** [comment]
Same issue. Getting `couldn't find DSO to load: libreactnativejni.so` with Android Gradle Plugin 3.3 and using Android App bundle. Even with RN 0.58 or 0.58.1, it doest not matter

**Comment by namvoeh** [comment]
@dulmandakh  `App Bundling support landed in 0.58. Please upgrade to 0.58.`
Can you share the changelog? I couldn't find it

**Comment by dulmandakh** [comment]
IIRC, it was cherry-picked in 0.58.4.
